improver_tests/developer_tools/__init__.py

```diff
@@ -0,0 +1,228 @@
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Set up fixtures for metadata interpreter tests"""
+
+from datetime import datetime
+
+import iris
+import numpy as np
+import pytest
+
+from improver.developer_tools.metadata_interpreter import MOMetadataInterpreter
+from improver.spotdata.build_spotdata_cube import build_spotdata_cube
+from improver.synthetic_data.set_up_test_cubes import (
+    construct_scalar_time_coords,
+    set_up_percentile_cube,
+    set_up_probability_cube,
+    set_up_variable_cube,
+)
+from improver.wxcode.utilities import weather_code_attributes
+
+
+@pytest.fixture(name="interpreter")
+def interpreter_fixture():
+    return MOMetadataInterpreter()
+
+
+def _update_blended_time_coords(cube):
+    """Replace forecast period and forecast reference time with blend time
+    on blended cubes (to be updated in setup utilities once standard is
+    fully implemented).  Modifies cube in place."""
+    blend_time = cube.coord("forecast_reference_time").copy()
+    blend_time.rename("blend_time")
+    cube.add_aux_coord(blend_time)
+    cube.remove_coord("forecast_period")
+    cube.remove_coord("forecast_reference_time")
+
+
+@pytest.fixture(name="probability_above_cube")
+def probability_above_fixture():
+    """Probability of air temperature above threshold cube from UKV"""
+    data = 0.5 * np.ones((3, 3, 3), dtype=np.float32)
+    thresholds = np.array([280, 282, 284], dtype=np.float32)
+    attributes = {
+        "source": "Met Office Unified Model",
+        "title": "Post-Processed UKV Model Forecast on 2 km Standard Grid",
+        "institution": "Met Office",
+        "mosg__model_configuration": "uk_det",
+    }
+    return set_up_probability_cube(data, thresholds, attributes=attributes)
+
+
+@pytest.fixture(name="blended_probability_below_cube")
+def probability_below_fixture():
```

Comment:
Could the fixture name and function name follow a standard naming convention? [words]_cube and [words]_fixture where words match exactly? It'll make them much easier to find.

Reply:
Maybe - but it'll also make them much harder to navigate in terms of what the imports mean.  Because the imports have to be in alphabetical order, if I add the relevant detail from the cube name to the pytest fixture name, and then rearrange this module alphabetically too, the "probability above" and "probability below" cubes will be defined miles from each other, and lose the logical link.  I would rather arrange everything alphabetically in terms of fixture name, and retain the detail in the cube names to aid interpretation of specific tests.

Reply:
You could rearrange the nouns such that "blended_probability_below_cube" becomes "probability_below_blended_cube".

Reply:
I could ... but I don't like how that reads at all.

Having added a conftest.py file, this has removed the need to import the fixtures directly and makes it less likely that developers will be confused by the lack of "translation" (the only place where fixture names are used, the cube name is directly defined there too).  So I think that makes it less crucial to match these names directly?  I've argued this on @tjtg 's comment below, as well - but if you both still feel strongly that I am wrong, I can change this on Monday.